api/column_family.cc

```diff
@@ -1017,12 +1017,13 @@ void set_column_family(http_context& ctx, routes& r, sharded<db::system_keyspace
         auto key = req->get_query_param("key");
         auto uuid = get_uuid(req->param["name"], ctx.db.local());
 
-        return ctx.db.map_reduce0([key, uuid] (replica::database& db) {
-            return db.find_column_family(uuid).get_sstables_by_partition_key(key);
+        return ctx.db.map_reduce0([key, uuid] (replica::database& db) -> future<std::unordered_set<sstring>> {
+            auto sstables = co_await db.find_column_family(uuid).get_sstables_by_partition_key(key);
+            auto names = sstables | boost::adaptors::transformed([] (auto s) { return s->get_filename(); });
+            co_return std::unordered_set<sstring>(names.begin(), names.end());
```

Comment:
This can be achieved more simply in a single pass with boost::copy_range<std::unordered_set<sstring>>(sstables | boost::adaptors::transformed ...)